connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala

```diff
@@ -212,11 +212,19 @@ object SchemaConverters extends Logging {
           ).toSeq
           fields match {
             case Nil =>
-              log.info(
-                s"Dropping ${fd.getFullName} as it does not have any fields left " +
-                "likely due to recursive depth limit."
-              )
-              None
+              if (protobufOptions.retainEmptyMessage) {
+                // Insert a dummy column to retain the empty message because
+                // spark doesn't allow empty struct type.
+                Some(
+                  StructType(StructField("_dummy_field_to_retain_empty_message", StringType) :: Nil)
```

Comment:
Better to use double underscores `__`, also it is quite long. How about `__dummy_field_in_empty_struct` ? Even my suggestion is bit long :).

Reply:
Fixed.